DesignSafe-CI/portal
# Replace "start date" filters with simple show/hide history toggle.

## Overview: ##

Users were confused by the default view and could not easily figure out
how to see what is happening right now in the facilities.  They were being
presented by default with the full history of all projects in all facilities,
sorted from newest to oldest, causing projects scheduled a year or more into
the future to be the first thing displayed when they load the page.

This change replaces the menu of "start date"-based filters entirely with 
a simple checkbox to show or hide projects that are already completed.  
It also reverses the default ordering from "newest first" to "oldest first".

Applying the new "hide history" filter and reversing the sort order by default 
on page load will mean users are now presented immediately with a view
of projects that is more in-line with expectations:

 * Projects in facilities currently are the first things they see, and
 * Upcoming projects appear the further down the page they go.

## PR Status: ##


## Related Jira tickets: ##

* [DES-XXX](https://jira.tacc.utexas.edu/browse/DES-XXX)

## Summary of Changes: ##

* Remove "start date" filter menu from filter bar component.
* Add checkbox to show or hide events with end date in the past.
* Add a default filter to the initial loadProjects() call when the page loads that activates this "hide history" filter.
* Reverse the default sort order from descending by "start date" to ascending.

## Testing Steps: ##
1. Load the app page (/nco)
2. Confirm the events are sorted by oldest "start date" first instead of newest upon initial page load.
3. Confirm the default view on initial page load is not showing events with an "end date" before today.
4. Confirm the show/hide history checkbox is working.


## UI Photos: ##

![nco-app-history-on](https://user-images.githubusercontent.com/5141273/146029533-34049be9-22c3-4df0-bb63-ac66e55cada5.png)

![nco-app-history-off](https://user-images.githubusercontent.com/5141273/146029577-44aca168-f71b-4d00-b268-8cbb8570ec26.png)

## Notes: ##

Events that have an empty "start date" field are being sorted at the top of the list.  This is not a code problem.  This is a data problem fixed in the database.  These entries just need to be updated with a start date to fall into their appropriate place in the list.

The code for how to setup a checkbox like this was borrowed from how the Data Depot file listings are doing it.
